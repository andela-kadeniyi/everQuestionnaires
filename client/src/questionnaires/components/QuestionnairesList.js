import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import Questionnaire from './Questionnaire';

const QuestionnairesList = ({ questionnaires }) => (
  <ul>
    { _.map(questionnaires, questionnaire => 
      <Questionnaire
        key={questionnaire.id}
        {...questionnaire}
      />
    )}
  </ul>
);

QuestionnairesList.propTypes = {
  questionnaires: PropTypes.objectOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired
  }).isRequired).isRequired
};

export default QuestionnairesList;