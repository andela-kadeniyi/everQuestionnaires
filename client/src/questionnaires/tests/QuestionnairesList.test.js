import React from 'react';
import { shallow } from 'enzyme';
import QuestionnairesList from '../components/QuestionnairesList';
import { formattedData } from './testData';

//initialize with props if none are given
const setup = (props = formattedData) => {
  
    const component = shallow(
      <QuestionnairesList questionnaires={props}  />
    );

  return {
    component
  };
}

describe('QuestionnairesList component', () => {
    const { component } = setup();
  
  it('renders self with questionnaires props', () => {
    expect(component.instance().props).toEqual({questionnaires: formattedData}); 
    expect(component).toBePresent();
  });
  
  it('renders a list of questionnaire links', () => {
    expect(component.find('ul').children().length).toEqual(Object.keys(formattedData).length);
    // const list = component.find('li');
    // //expect(component.props('questionnaires').children).toEqual(formattedData);
    // expect(component.find('li').map(n => console.log(n.text())));
    // expect(list.at(1)).toEqual(formattedData["2"].id);
  });
});