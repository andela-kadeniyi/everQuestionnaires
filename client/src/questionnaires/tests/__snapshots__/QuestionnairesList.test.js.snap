// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnairesList component renders a list of questionnaire links 1`] = `
<ul>
  <li>
    <Link
      replace={false}
      to="questionnaires/1"
    >
      Lord of the Rings
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="questionnaires/2"
    >
      Lord of the Flies
    </Link>
  </li>
</ul>
`;

exports[`QuestionnairesList component renders self with questionnaires props 1`] = `
<ul>
  <li>
    <Link
      replace={false}
      to="questionnaires/1"
    >
      Lord of the Rings
    </Link>
  </li>
  <li>
    <Link
      replace={false}
      to="questionnaires/2"
    >
      Lord of the Flies
    </Link>
  </li>
</ul>
`;
